### PR TITLE
Update tabular data examples

### PR DESCRIPTION
**Pull request by npentrel**

### CODE REVIEW

File: src/viam/app/data_client.py [hunk 1]
```diff
@@ -255,10 +255,14 @@ async def tabular_data_by_sql(self, organization_id: str, sql_query: str) -> Lis
...
     async def tabular_data_by_mql(self, organization_id: str, mql_binary: List[bytes]) -> List[Dict[str, ValueTypes]]:
         """Obtain unified tabular data and metadata, queried with MQL.
+        You must install the `pymongo` package and `import bson` to use the `tabular_data_by_mql` method.
 
         ::
 
-            data = await data_client.tabular_data_by_mql(org_id="<your-org-id>", mql_binary=[<mql-bytes-1>, <mql-bytes-2>])
+            data = await data_client.tabular_data_by_mql(org_id="<your-org-id>", mql_binary=[
+                bson.encode({'$match': {'location_id': '<location-id>'} }),
+                bson.encode({ "$limit": 5 })
```
Comment (npentrel) [inline src/viam/app/data_client.py:264]:
```suggestion
                bson.encode({ '$match': { 'location_id': '<location-id>' } }),
                bson.encode({ "$limit": 5 })
```

File: src/viam/app/data_client.py [hunk 2]
```diff
@@ -255,10 +255,14 @@ async def tabular_data_by_sql(self, organization_id: str, sql_query: str) -> Lis
 
     async def tabular_data_by_mql(self, organization_id: str, mql_binary: List[bytes]) -> List[Dict[str, ValueTypes]]:
         """Obtain unified tabular data and metadata, queried with MQL.
+        You must install the `pymongo` package and `import bson` to use the `tabular_data_by_mql` method.
```
Comment (njooma) [inline src/viam/app/data_client.py:258]: Is this accurate? I don't think this is strictly necessary -- you can use any bson serialization library. I would change this to be more of a suggestion, something like:

> `mql_binary` is an array of encoded `bson` queries. You can encode your `bson` queries using a library like `pymongo`.

**Review by njooma — CHANGES_REQUESTED** [review]

File: src/viam/app/data_client.py (hunk 2 above)
Comment (npentrel) [reply src/viam/app/data_client.py:258]: well you cannot use the regular bson library that I first had installed. https://github.com/hyperopt/hyperopt/issues/547

Comment (njooma) [reply src/viam/app/data_client.py:258]: What happens when you try to use the `bson` library? I was just able to do this:

```python3
import bson

bson.dumps({ '$match': { 'location_id': '<location-id>' }})
```

And that outputs a valid bson binary. I'm curious as to the functionality you are seeing.

Comment (npentrel) [reply src/viam/app/data_client.py:258]: that does indeed work. You can't call encode on that. I think that's what I was trying before and it has just a varietey of `encode_*` methods

Comment (npentrel) [inline src/viam/app/data_client.py:258]:
```suggestion
        You must `import bson` to create valid bson binary objects as input for the `tabular_data_by_mql` method.
```

File: src/viam/app/data_client.py [hunk 3]
```diff
@@ -255,10 +255,14 @@ async def tabular_data_by_sql(self, organization_id: str, sql_query: str) -> Lis
...
 
         ::
 
-            data = await data_client.tabular_data_by_mql(org_id="<your-org-id>", mql_binary=[<mql-bytes-1>, <mql-bytes-2>])
+            data = await data_client.tabular_data_by_mql(org_id="<your-org-id>", mql_binary=[
+                bson.encode({ '$match': { 'location_id': '<location-id>' } }),
+                bson.encode({ "$limit": 5 })
```
Comment (npentrel) [inline src/viam/app/data_client.py:264]:
```suggestion
                bson.dumps({ '$match': { 'location_id': '<location-id>' } }),
                bson.dumps({ "$limit": 5 })
```

**Review by stuqdog — APPROVED** [review]
Looks reasonable to me though I don't have a ton of experience with bson. I assume this was all tested and works?

**Comment by npentrel** [comment]
tested both, yes

File: src/viam/app/data_client.py [hunk 4]
```diff
@@ -255,10 +255,14 @@ async def tabular_data_by_sql(self, organization_id: str, sql_query: str) -> Lis
 
     async def tabular_data_by_mql(self, organization_id: str, mql_binary: List[bytes]) -> List[Dict[str, ValueTypes]]:
         """Obtain unified tabular data and metadata, queried with MQL.
+        You must `import bson` to create valid bson binary objects as input for the `tabular_data_by_mql` method.
```
Comment (njooma) [inline src/viam/app/data_client.py:258]: Again, I don't want to prescribe a library, particularly given that the `bson` library hasn't been updated in 3 years. Further, a library isn't even technically required -- you could encode the bson by hand.

My suggestion would be to remove the `must`, and then provide an example of using a library:

```python3
async def tabular_data_by_mql(...):
    """Obtain unified tabular data and metadata, queried with MQL.
    `mql_binary` is an array of encoded `bson` queries. You can encode your bson queries using a library like `pymongo` or `bson`.

    ::
        # using bson 
        import bson
        data = ...(..., mql_binary=[bosn.dumps(...)])

        # using pymongo 
        import bson
        data = ...(..., mql_binary=[bosn.encode(...)])
"""

**Review by njooma — CHANGES_REQUESTED** [review]

File: src/viam/app/data_client.py (hunk 4 above)
Comment (npentrel) [reply src/viam/app/data_client.py:258]: You didn't actually say that before. I'm ok using your proposal.

File: src/viam/app/data_client.py [hunk 5]
```diff
@@ -255,15 +255,26 @@ async def tabular_data_by_sql(self, organization_id: str, sql_query: str) -> Lis
 
     async def tabular_data_by_mql(self, organization_id: str, mql_binary: List[bytes]) -> List[Dict[str, ValueTypes]]:
         """Obtain unified tabular data and metadata, queried with MQL.
-
+
```
Comment (npentrel) [inline src/viam/app/data_client.py:258]:
```suggestion

```

File: src/viam/app/data_client.py [hunk 6]
```diff
@@ -255,15 +255,25 @@ async def tabular_data_by_sql(self, organization_id: str, sql_query: str) -> Lis
 
     async def tabular_data_by_mql(self, organization_id: str, mql_binary: List[bytes]) -> List[Dict[str, ValueTypes]]:
         """Obtain unified tabular data and metadata, queried with MQL.
-
         ::
+            # using bson
+            import bson
+            data = await data_client.tabular_data_by_mql(org_id="<your-org-id>", mql_binary=[
```
Comment (npentrel) [inline src/viam/app/data_client.py:261]:
```suggestion
            tabular_data = await data_client.tabular_data_by_mql(org_id="<your-org-id>", mql_binary=[
```

File: src/viam/app/data_client.py [hunk 7]
```diff
@@ -255,15 +255,25 @@ async def tabular_data_by_sql(self, organization_id: str, sql_query: str) -> Lis
...
+                bson.dumps({ '$match': { 'location_id': '<location-id>' } }),
+                bson.dumps({ "$limit": 5 })
+            ])
 
-            data = await data_client.tabular_data_by_mql(org_id="<your-org-id>", mql_binary=[<mql-bytes-1>, <mql-bytes-2>])
+            # using pymongo
+            import bson
+            data = await data_client.tabular_data_by_mql(org_id="<your-org-id>", mql_binary=[
```
Comment (npentrel) [inline src/viam/app/data_client.py:268]:
```suggestion
            tabular_data = await data_client.tabular_data_by_mql(org_id="<your-org-id>", mql_binary=[
```

File: src/viam/app/data_client.py [hunk 8]
```diff
@@ -255,15 +255,25 @@ async def tabular_data_by_sql(self, organization_id: str, sql_query: str) -> Lis
...
+                bson.encode({ '$match': { 'location_id': '<location-id>' } }),
+                bson.encode({ "$limit": 5 })
+            ])
 
 
         Args:
             organization_id (str): The ID of the organization that owns the data.
-            mql_binary (List[bytes]):The MQL query to run as a list of BSON documents.
+            mql_binary (List[bytes]):The MQL query to run as a list of BSON queries. You can encode your bson queries using a library like `pymongo` or `bson`.
```
Comment (npentrel) [inline src/viam/app/data_client.py:276]:
```suggestion
            mql_binary (List[bytes]): The MQL query to run as a list of BSON queries. You can encode your bson queries using a library like `pymongo` or `bson`.
```

File: src/viam/app/data_client.py [hunk 9]
```diff
@@ -255,15 +255,25 @@ async def tabular_data_by_sql(self, organization_id: str, sql_query: str) -> Lis
 
     async def tabular_data_by_mql(self, organization_id: str, mql_binary: List[bytes]) -> List[Dict[str, ValueTypes]]:
         """Obtain unified tabular data and metadata, queried with MQL.
-
         ::
```
Comment (npentrel) [inline src/viam/app/data_client.py:258]:
```suggestion

        ::
```

**Review by njooma — APPROVED** [review]

File: src/viam/app/data_client.py [hunk 10]
```diff
@@ -257,13 +257,24 @@ async def tabular_data_by_mql(self, organization_id: str, mql_binary: List[bytes
...
+            tabular_data = await data_client.tabular_data_by_mql(org_id="<your-org-id>", mql_binary=[
+                bson.encode({ '$match': { 'location_id': '<location-id>' } }),
+                bson.encode({ "$limit": 5 })
+            ])
 
 
         Args:
             organization_id (str): The ID of the organization that owns the data.
-            mql_binary (List[bytes]):The MQL query to run as a list of BSON documents.
+            mql_binary (List[bytes]): The MQL query to run as a list of BSON queries. You can encode your bson queries using a library like `pymongo` or `bson`.
```
Comment (npentrel) [inline src/viam/app/data_client.py:277]:
```suggestion
            mql_binary (List[bytes]): The MQL query to run as a list of BSON queries. You can encode your bson queries using a library like
            `pymongo` or `bson`.
```

**Pull request merged**